clipper_admin/clipper_admin/clipper_admin.py

```diff
@@ -0,0 +1,995 @@
+        ``model_data_path`` copied into the image. The Dockerfile that gets generated to build the image
+        is equivalent to the following::
+
+            FROM <base_image>
+            COPY <model_data_path> /model/
+
+        The newly built image is then pushed to the specified container registry. If no container registry
+        is specified, the image will be pushed to the default DockerHub registry. Clipper will tag the
+        newly built with the tag [<registry>]/<name>:<version>, and if a container registry is provided,
+
+        2. Registers and deploys a model with the specified metadata using the newly built container as the
+        Docker image for that model. See :py:meth:`clipper.ClipperConnection.deploy_model` for more details
+        on model deployment.
+        
+        Parameters
+        ----------
+        name : str
+            The name of the deployed model
+        version : str
+            The version to assign this model. Versions must be unique on a per-model
+            basis, but may be re-used across different models.
+        input_type : str
+            The type of the request data this endpoint can process. Input type can be
+            one of "integers", "floats", "doubles", "bytes", or "strings". See the
+            `User Guide <http://clipper.ai/user_guide/#input-types>`_ for more details
+            on picking the right input type for your application.
+        model_data_path : str
+            A path to a local directory. The contents of this directory will be recursively copied into the
+            Docker container.
+        base_image : str
+            The base Docker image to build the new model image from. This
+            image should contain all code necessary to run a Clipper model
+            container RPC client.
+        labels : list(str), optional
+            A list of strings annotating the model. These are ignored by Clipper
+            and used purely for user annotations.
+        container_registry : str, optional
+            The Docker container registry to push the freshly built model to. Note
+            that if you are running Clipper on Kubernetes, this registry must be accesible
+            to the Kubernetes cluster in order to fetch the container from the registry.
+        num_replicas : int, optional
+            The number of replicas of the model to create. The number of replicas
+            for a model can be changed at any time with
+            :py:meth:`clipper.ClipperConnection.set_num_replicas`.
+
+        Raises
+        ------
+        :py:exc:`clipper.UnconnectedException`
+        :py:exc:`clipper.ClipperException`
+        """
+
+        if not self.connected:
+            raise UnconnectedException()
+        version = str(version)
+
+        _validate_versioned_model_name(name, version)
+
+        with open(model_data_path + '/Dockerfile', 'w') as f:
```

Comment:
As far as I can tell, couldn't we still potentially be overwriting a dockerfile here? There was a comment indicating that this was fixed, but I'm not sure:

Comment:
"rename the Dockerfile to a random string to avoid overwriting existing dockerfiles and set the `dockerfile` argument in `docker_client.images.build`"

Reply:
Yeah that didn't work with the Python docker SDK. I added a check and raise an exception if a Dockerfile already exists at that location.